### ISSUE TITLE
etcd/raft Adding a node to an existing cluster  

### ISSUE THREAD

**Issue by achraf17** [etcd-io/etcd#13306]
etcd version: v3.5.0

Based on the documentation of the [readme](https://github.com/etcd-io/etcd/blob/v3.5.0/raft/README.md#usage) for adding a new node to the cluster, use `raft.StartNode(c, nil)`. 
But based on the [code](https://github.com/etcd-io/etcd/blob/v3.5.0/raft/node.go#L219), it will panic with an error of `"no peers given; use RestartNode instead"` as `len(peers) == 0` will be true.

Is there any alternative ways to add a node to an existing cluster ?

**Comment by qdongxu** [comment]
Can you describe what you did step by step, allowing others to reproduce the course. e.g. what are the existing clusters' topologies and configurations, how you set the parameters for the new node,  how the code writing, and what is the exact panic stacks?  It's not enough to describe as "I following the readme".

**Comment by achraf17** [comment]
Hey @qdongxu,
I had a network with 3 nodes running in a cluster. To add a new node, i did:
Step 1: I called `ProposeConfChange(context.TODO(), cc)` with cc being
```
 raftpb.ConfChange{
          ID: id,
	  Type:    raftpb.ConfChangeAddNode,
	  NodeID:  uint64(raftId),
	  Context: address.toBytes(),
}
```
Step2: then i started the new node by calling  `raft.StartNode(c, nil)` with c being
```
Raft.Config{
		Applied:       lastAppliedIndex,
		ID:            uint64(id),
		ElectionTick:  10, 
		HeartbeatTick: 1,
		Storage:       raftStorage,
		CheckQuorum: false,
		MaxSizePerMsg: 4096, 
		MaxInflightMsgs: 256,
	}
```
Just after calling the `raft.StartNode` with the `nil` value in the peers param, the node stops with no stack error on my code but when I follow the code implementation of `raft.StartNode` I found the [panic error](https://github.com/etcd-io/etcd/blob/v3.5.0/raft/node.go#L220).
Let me know if more info is required.
